measuretrackquality.py

```diff
@@ -0,0 +1,631 @@
+CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+Author: Dan Ruderman (ruderman@usc.edu)
+"""
+
+from collections import Counter, defaultdict
+import itertools
+import logging
+import numpy
+import re
+import scipy.interpolate
+
+import cellprofiler.measurement as cpmeas
+import cellprofiler.module as cpm
+import cellprofiler.setting as cps
+from cellprofiler.measurement import M_NUMBER_OBJECT_NUMBER
+from cellprofiler.modules import trackobjects
+from cellprofiler_plugins.setting import MeasurementMultiChoiceForCategory
+
+TRAM_MIN_TIME_POINTS = 6 # minimum number of time points to calculate TrAM
+
+__doc__ = """
+<b>TrackQuality</b> provides tracking quality metrics. TrAM (Tracking
+Aberration Measure) is based on temporal smoothness of features measured
+across each object's trajectory.
+<hr>
+This module must be placed downstream of a module that identifies objects
+(e.g., <b>IdentifyPrimaryObjects</b>) and a <b>TrackObjects</b> that tracks
+them. There must be at least {min_tp} frames to perform a TrAM analysis. The TrAM
+statistic reflects how typical the maximum deviation from smooth time series
+a chosen set of measurements are. Typical fluctuations are determined from
+measurement differences in adjacent time points of objects whose trajectories
+are complete and without splitting events.
+<p>
+<h4>Available measurements</h4>
+<ul>
+<li><i>TrAM:</i> The TrAM value for the trajectory. Values near 1 are typical
+for a good trajectory. Large values (typically 3 or higher) are more likely
+to correspond to aberrant tracks. The value <i>None</i> is assigned to objects
+with partial tracks or those for whom <i>Is_Parent</i> is 1. A histogram of
+all computed TrAM values is displayed to help define a cutoff.</li>
+<li><i>Labels:</i> Each tracked item that has a lineage from the first to the
+last frame is assigned a TrAM label on the last frame. If the final object
+does not arise from a split during tracking, then it has this same unique label
+for its entire track. That label is not assigned to any other objects. But if
+an object arises from a split, then its ancestor object(s) will be assigned
+multiple labels (combined from its progeny). These labels are separated by
+a "|" symbol.</li>
+<li><i>Is_Parent:</i> If the object splits into daughters during its track then 
+flag will be 1, and <i>Labels</i> will be a list of two or more labels.
+Otherwise it is 0 and <i>Labels</i> is a list containing one label.</li>
+<li><i>Split_Trajectory:</i> If the object arose from an ancestor whose trajectory
+split, then this value is 1. Otherwise it is 0.</li>
+</ul>
+<p>
+<h4>Publications</h4>
+Patsch, K <i>et al.</i>, <a href=https://www.nature.com/articles/srep34785>Single cell
+dynamic phenotyping</a>, <i>Scientific Reports</i> 6:34785 (2016)
+<p>
+<h4>Author</h4>
+Dan Ruderman (ruderman@usc.edu)
+""".format(**{"min_tp":TRAM_MIN_TIME_POINTS})
+
+# Features to add:
```

Comment:
@DanRuderman I'm curious if you're planning on implementing these before this module gets merged into master?